tsdb/engine/tsm1/reader.go

```diff
@@ -796,6 +810,45 @@ func (f *fileAccessor) readBytes(entry *IndexEntry, b []byte) ([]byte, error) {
 	return b[4:n], nil
 }
 
+// getChecksums returns a map[string]bool in the form of blockId: valid (writtenChecksum == calulcatedChecksum)
+func (f *fileAccessor) getChecksums(key string) (map[string]bool, error) {
```

Comment:
This still doesn't seem like the best way to handle the checksums.  Seems like getting the checksum should be part of the `BlockIterator.Read` call here: https://github.com/influxdata/influxdb/blob/master/tsdb/engine/tsm1/reader.go#L80

Given a TSM file, you can iterate over each block sequentially and compare the checksum stored in the file vs re-calculating it based on the raw bytes returned from that call.